File: hw5b/2/Fruit.cpp
#include <iostream>
#include <string>
#include "Fruit.hpp"

Fruit::Fruit(time_t ripe, std::string name) {
    std::cout << "Fruit::Fruit()\n";
    this->ripe = ripe;
    this->name = name;
}

void Fruit::ripeTime() {
    std::cout
        << "You can eat it on "
        << time(&this->ripe)
        << std::endl;
}

File: hw5b/1/Fruit.java
import java.util.Date;

public class Fruit extends Food implements Comparable<Fruit> {
    protected Date ripe;

    public Fruit(Date r, String n) {
        this.ripe = r;
        this.name = n;
    }

    public void ripeTime() {
        System.out.println("You can eat it on " + this.ripe);
    }

    @Override
    public String toString() {
        return String.format("%-12s (%s)", this.name, this.ripe);
    }

    @Override
    public int compareTo(Fruit other) {
        return this.ripe.compareTo(other.ripe);
    }
}
